[OSS] feat(cli): add initial peering cli commands (#14423)
hashicorp/consul: website/content/commands/peering/generate-token.mdx

---
layout: commands
page_title: 'Commands: Peering Generate Token'
---

# Consul Peering Generate Token

Command: `consul peering generate-token`

Corresponding HTTP API Endpoint: [\[POST\] /v1/peering/token](/api-docs/peering#generate-a-peering-token)

> The `peering generate-token` generates a peering token.

Reviewer: ```suggestion
The `peering generate-token` generates a peering token. The token is base 64-encoded string containing the token details. 
```
Saw this bit about the encoding in the main docs and thought it made sense to put here, too.

This token should be transferred to the other cluster being peered and consumed using [`consul peering establish`](/commands/peering/establish).

Re-generating a token for a given name will not interrupt any active connection, but will
invalidate any unused token for that name.

The table below shows this command's [required ACLs](/api#authentication).

| ACL Required |
| ------------ |
| `peering:write`  |

## Usage

Usage: `consul peering generate-token [options] -name <peer name>`

#### Command Options

> - `-name=<string>` - (Required) The name of the peer.
> This name will only be used locally to refer to the connection.

Reviewer: ```suggestion
- `-name=<string>` - (Required) Specifies a local name for the cluster that the token is intended for. The `name` is only used to identify the connection with the peer. 
```
I would also add the information about duplicate names when generating a new token here because I don't think the reader will have the context until they get to this flag.

> - `-meta=<string>=<string>` - Metadata to associate with the peering,
> formatted as key=value.
> This flag may be specified multiple times to set multiple metadata fields.

Reviewer: ```suggestion
- `-meta=<string>=<string>` - Specifies key/value pairs to associate with the peering connection token in `-meta="key"="value"` format. You can use the flag multiple times to set multiple metadata fields.
```

> - `-server-external-addresses=<string>[,string,...]` - A list of addresses to put into
> the generated token, formatted as a comma-separate list.
> This could be used to specify load balancer(s) or external IPs to reach the servers from
> the dialing side, and will override any server addresses obtained from the "consul" service.

Reviewer: ```suggestion
- `-server-external-addresses=<string>[,string,...]` - Specifies a comma-separated list of addresses to put into the generated token. You can specify one or more load balancers or external IPs that connect to Consul servers from the cluster that initiates the connection. As a result, Consul overrides any server addresses obtained from the `consul` service with this value.
```
Correct?

Author: To be consistent with my other suggestion.
```suggestion
- `-server-external-addresses=<string>[,string,...]` - Specifies a comma-separated list of addresses to put into the generated token. You can specify one or more load balancers or external IPs that route external traffic to this cluster's Consul servers. 
```

- `-format={pretty|json}` - Command output format. The default value is `pretty`.

#### Enterprise Options

@include 'http_api_partition_options.mdx'

#### API Options

@include 'http_api_options_client.mdx'

## Examples

> To generate a peering token for cluster, which will be called "cluster-02":

Reviewer: ```suggestion
The following example generates a peering token for a cluster called "cluster-02":
```

```shell-session hideClipboard
$ consul peering generate-token -name cluster-02
eyJDQSI6bnVs...5Yi0wNzk5NTA1YTRmYjYifQ==
```

### Using a Load Balancer for Consul Servers

> To generate a token for a cluster where servers are proxied by a load balancer:

Reviewer: ```suggestion
The following example generates a token for a cluster where servers are proxied by a load balancer:
```
Might need to take another stab at the description for the s-e-add flag
